apps/angular/projection/src/app/ui/card/card.component.html

```diff
@@ -1,27 +1,20 @@
 <div
   class="border-2 border-black rounded-md p-4 w-fit flex flex-col gap-3"
   [class]="customClass">
-  <img
-    *ngIf="type === CardType.TEACHER"
-    src="assets/img/teacher.png"
-    width="200px" />
-  <img
-    *ngIf="type === CardType.STUDENT"
-    src="assets/img/student.webp"
-    width="200px" />
+  <ng-content select="img"></ng-content>
 
   <section>
     <app-list-item
       *ngFor="let item of list"
-      [name]="item.firstname"
+      [name]="item[itemNameKey]"
       [id]="item.id"
-      [type]="type">
+      (onDelete)="onDelete.emit($event)">
```

Comment:
as you mention inside your description, you should use ngTemplateOutlet here. I should have asked for more difference between cards because in this example, it's too simple to concatenate string to create a correct name.
But what happens if you need to add an icon. or anything else. Your shared component became less customizable.